apps/rush-lib/src/logic/DependencySpecifier.ts

```diff
@@ -4,6 +4,56 @@
 import npmPackageArg = require('npm-package-arg');
 import { InternalError } from '@rushstack/node-core-library';
 
+/**
+ * The parsed format of a provided version specifier.
+ */
+export const enum SpecifierType {
```

Comment:
Why a `const enum`? And the name should probably be a little more specific.

```suggestion
export enum DependencySpecifierType {
```